be/osoc/common/filters.py

```diff
@@ -74,13 +75,16 @@ class EmailDateTimeFilter(filters.BaseFilterBackend):
     """
     def filter_queryset(self, request, queryset, view):
         try:
-            if (date := request.query_params.get('date')) is not None:
-                return queryset.filter(time__date=date)
-            if (before := request.query_params.get('before')) is not None:
-                queryset = queryset.filter(time__lt=before)
-            if (after := request.query_params.get('after')) is not None:
-                queryset = queryset.filter(time__gt=after)
-        except ValidationError:
+            date = request.query_params.get('date')
+            if date is not None:
+                return queryset.filter(time__date=string_to_datetime_tz(date))
+            before = request.query_params.get('before')
+            if before is not None:
+                queryset = queryset.filter(time__lt=string_to_datetime_tz(before))
+            after = request.query_params.get('after')
+            if after is not None:
+                queryset = queryset.filter(time__gt=string_to_datetime_tz(after))
+        except ValueError:
             # return default queryset when a validationerror is raised
```

Comment:
Shouldn't `validationerror` be `ValueError`? Not that it matters that much but now the comment does not make sense.